cli/azd/pkg/project/project_config.go

```diff
@@ -282,6 +284,10 @@ func LoadProjectConfig(projectPath string) (*ProjectConfig, error) {
 		return nil, fmt.Errorf("parsing project file: %w", err)
 	}
 
+	if projectConfig.Metadata != nil {
+		telemetry.SetUsageAttributes(fields.StringHashed(fields.TemplateIdKey, projectConfig.Metadata.Template))
```

Comment:
Again, this feels like something the `ProjectConfig` shouldn't be concerned about and would be better off pulling out a runtime during telemetry collection time (via middleware maybe)

Reply:
Let's discuss on https://github.com/Azure/azure-dev/pull/1302#discussion_r1060878331